compiler/crates/relay-lsp/src/rename/mod.rs

```diff
@@ -0,0 +1,616 @@
+
+                    Ok(RenameRequest::new(
+                        RenameKind::OperationDefinition,
+                        location,
+                    ))
+                }
+                _ => Err(LSPRuntimeError::ExpectedError),
+            }
+        }
+        Feature::DocblockIr(_) => Err(LSPRuntimeError::ExpectedError),
+    }
+}
+
+pub fn get_locations_for_rename(
+    rename_request: RenameRequest,
+    program: &Program,
+) -> LSPRuntimeResult<Vec<IRLocation>> {
+    match rename_request.kind {
+        RenameKind::Identifier => Ok(vec![rename_request.location]),
+        RenameKind::OperationVariable {
+            variable_name,
+            operation_name,
+        } => VariableUsageFinder::get_variable_locations(program, operation_name, variable_name),
+        RenameKind::FragmentArgument {
+            fragment_name,
+            argument_name,
+        } => Ok(FragmentArgumentFinder::get_argument_locations(
+            program,
+            fragment_name,
+            argument_name,
+        )),
+        RenameKind::OperationDefinition => Ok(vec![rename_request.location]),
+        RenameKind::FragmentDefinitionOrSpread { fragment_name } => {
+            Ok(FragmentFinder::get_fragment_usages(program, fragment_name))
+        }
+    }
+}
+
+fn get_rename_request_by_definition(
+    variable_name: StringKey,
+    variable_location: IRLocation,
+    definition: &ExecutableDefinition,
+) -> LSPRuntimeResult<RenameRequest> {
+    match definition {
+        ExecutableDefinition::Fragment(fragment_definition) => {
+            let rename_kind = if fragment_definition
+                .variable_definitions
+                .as_ref()
+                .map_or(false, |variables| {
+                    variables.items.iter().any(|v| v.name.name == variable_name)
+                }) {
+                RenameKind::FragmentArgument {
+                    fragment_name: fragment_definition.name.value,
+                    argument_name: variable_name,
+                }
+            } else if fragment_definition
+                .directives
+                .named(*ARGUMENTDEFINITIONS_DIRECTIVE)
+                .map_or(false, |directive| {
+                    directive.arguments.as_ref().map_or(false, |args| {
+                        args.items.iter().any(|v| v.name.value == variable_name)
+                    })
+                })
+            {
+                RenameKind::FragmentArgument {
+                    fragment_name: fragment_definition.name.value,
+                    argument_name: variable_name,
+                }
+            } else {
+                // The variable not being defined in the fragment definition
+                // doens't mean it's not defined at all. There's also the
+                // possibility of it being a global operation variable.
+                // For now we just rename the identifier without modifying
+                // any other parts.
+                RenameKind::Identifier
```

Comment:
This seems like it might be more confusing than helpful. If I click a global variable and select rename, and it only renames the thing I was clicking, I would probably be confused since I was expecting the global variable to get renamed. Should we just skip offering the refactor in this case?

Reply:
Yes, I agree. I'll remove it.

Reply:
Removed :)